numbagg/decorators.py

```diff
@@ -596,7 +601,23 @@ def _create_gufunc(self, target):
         return vectorize(self.func)
 
 
-def _is_in_thread_pool():
+def _is_in_unsafe_thread_pool() -> bool:
     current_thread = threading.current_thread()
     # ThreadPoolExecutor threads typically have names like 'ThreadPoolExecutor-0_1'
-    return "ThreadPoolExecutor" in current_thread.name
+    return current_thread.name.startswith(
+        "ThreadPoolExecutor"
+    ) and _thread_backend() in ["workqueue" or None]
```

Comment:
`["workqueue" or None]` == `["workqueue"]`

Reply:
> `["workqueue" or None]` == `["workqueue"]`

WDYM?

To the extent the proposal is just list -> set — fine , would take a PR...

Reply:
I think he means `_thread_backend() in ["workqueue", None]` the current one doesn't check `None`

Reply:
Oh!! Gosh, can't believe I missed that, and on the second look too. Fixed in #333